web/api/formatters/rrd2json.c

```diff
@@ -217,189 +253,222 @@ int rrdset2anything_api_v1(
         , time_t *latest_timestamp
         , struct context_param *context_param_list
         , char *chart_label_key
+        , long stats
 ) {
 
-    if (context_param_list && !(context_param_list->flags & CONTEXT_FLAGS_ARCHIVE))
-        st->last_accessed_time = now_realtime_sec();
+    int i = -1;
+    int last = rrdr_find_last(stats);
+    long step = 1;
 
-    RRDR *r = rrd2rrdr(st, points, after, before, group_method, group_time, options, dimensions?buffer_tostring(dimensions):NULL, context_param_list);
-    if(!r) {
-        buffer_strcat(wb, "Cannot generate output with these parameters on this chart.");
-        return HTTP_RESP_INTERNAL_SERVER_ERROR;
+    if(last == -1){
+        rrdr_open_json(wb);
     }
 
-    RRDDIM *temp_rd = context_param_list ? context_param_list->rd : NULL;
+    do{
+        if(i != -1){
+            if((stats & step)){
+                group_method = step;
+                step <<= 1;
+            }else{
+                step <<= 1;
+                //points = 0;
+                i++;
+                continue;
+            }
+        }
+        info("Erdem: %d, %d", last, i);
```

Comment:
@erdem2000 in normal situation I add debug messages while I am developing, and these debug messages have a specific message that I use `grep` to find, I am beginning to believe you also follow this development pattern. :thinking:

Reply:
:) Sorry for missing debug message, it will be removed.

Reply:
No problem, I also had to create commits to remove my debug messages in the past :rofl: .